src/sdk/pycli/nnicli/nni_client.py

```diff
@@ -92,48 +72,384 @@ def _create_process(cmd):
+
+class NNITrialJob:
+    """
+    NNITrialJob stores the information of a trial job.
+
+    Parameters
+    ----------
+    json_obj:
+        json object that stores the hyper parameters
+
+    Attributes
+    ----------
+    id:
+        trial id
+    status:
+        job status
+    hyperParameters:
+        see `NNITrialHyperParameters`
+    logPath:
+        log path
+    startTime:
+        job start time (timestamp)
+    endTime:
+        job end time (timestamp)
+    finalMetricData:
+        see `NNITrialMetricData`
+    parameter_index:
+        parameter index
+    """
+    def __init__(self, json_obj):
+        self.id = None
+        self.status = None
+        self.hyperParameters = None
+        self.logPath = None
+        self.startTime = None
+        self.endTime = None
+        self.finalMetricData = None
+        self.stderrPath = None
+        for key in json_obj.keys():
+            self.__setattr__(key, json_obj[key])
+        if self.hyperParameters:
+            self.hyperParameters = [NNITrialHyperParameters(json.loads(e)) for e in self.hyperParameters]
+        if self.finalMetricData:
+            self.finalMetricData = [NNITrialMetricData(e) for e in self.finalMetricData]
+
+    def __repr__(self):
+        return "NNITrialJob(id:{} status:{} hyperParameters:{} logPath:{} startTime:{} endTime:{} finalMetricData:{} stderrPath:{})" \
+            .format(self.id, self.status, self.hyperParameters, self.logPath,
+                    self.startTime, self.endTime, self.finalMetricData, self.stderrPath)
+
+class NNIExperiment:
+    def __init__(self):
+        self.endpoint = None
+        self.exp_id = None
+        self.port = None
+
+    def _exec_command(self, cmd, port=None):
+        if self.endpoint is not None:
+            raise RuntimeError('This instance has been connected to an experiment.')
+        if _create_process(cmd) != 0:
+            raise RuntimeError('Failed to establish experiment, please check your config.')
+        else:
+            if port:
+                self.port = port
+            else:
+                self.port = 8080
+            self.endpoint = 'http://localhost:{}'.format(self.port)
+            self.exp_id = self.get_experiment_profile()['id']
+
+    def start_experiment(self, config_file, port=None, debug=False):
+        """
+        Start an experiment with specified configuration file and connect to it.
+
+        Parameters
+        ----------
+        config_file: str
+            path to the config file
+        port: int
+            the port of restful server, bigger than 1024
+        debug: boolean
+            set debug mode
+        """
+        cmd = 'nnictl create --config {}'.format(config_file).split(' ')
+        if port:
+            cmd += '--port {}'.format(port).split(' ')
+        if debug:
+            cmd += ['--debug']
+        self._exec_command(cmd, port)
```

Comment:
Better if this is a classmethod that creates and returns a new experiment object.

Reply:
it is okay to write in this way :)